### ISSUE TITLE
TypeError in package_control/text.py

### ISSUE THREAD

**Issue by seanknox** [wbond/package_control#939]
1. This is with commit d3bc04fd5c0c4bdb476a3ea798e6815508dc9bac.
2. How is this component and other package control components tested? I've seen a lot of broken code in master—and I'd love to improve test coverage.
```
reloading Packages/User/Preferences.sublime-settings
Package Control: Installing 1 missing dependency
Exception in thread Thread-5:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "/Users/seanknox/Library/Application Support/Sublime Text 3/Packages/Package Control/package_control/automatic_upgrader.py", line 113, in run
    self.install_missing()
  File "/Users/seanknox/Library/Application Support/Sublime Text 3/Packages/Package Control/package_control/automatic_upgrader.py", line 146, in install_missing
    console_write(u'Installed missing dependency %s' % dependency, True)
  File "/Users/seanknox/Library/Application Support/Sublime Text 3/Packages/Package Control/package_control/console_write.py", line 34, in console_write
    string = text.format(str_cls(string), params, strip=strip, indent=indent)
  File "/Users/seanknox/Library/Application Support/Sublime Text 3/Packages/Package Control/package_control/text.py", line 57, in format
    output = output % params
TypeError: not all arguments converted during string formatting
```

**Comment by FichteFoll** [comment]
Covered by #934.

**Comment by seanknox** [comment]
Thanks @FichteFoll—still would like my second question about tests and test coverage answered.

**Comment by wbond** [comment]
@seanknox Tests are currently located at https://github.com/wbond/package_control/tree/master/package_control/tests. Currently they are focused on fetching various package metadata, but I don't have any opposition to adding more to cover other areas.

You'll need to enable the tests via the setting https://packagecontrol.io/docs/settings#settings-enable_tests. Also note that in order to not hit the GitHub API limit, you'll need to register an app and get a client id and secret and place them in https://github.com/wbond/package_control/blob/master/package_control/tests/__init__.py#L13-L14.

Moving forward we will probably want to split tests so that there is one command to run remote tests (slow, integration testing which require the GitHub credentials) with self-contained integration or unit tests.

**Comment by wbond** [comment]
This specific bug should be fixed by da829ffedde5874cc7514012491c0eb67cdcd218.

In terms of the tests, this bug is specific enough and deep enough into the innards of Sublime Text and the package loading that creating a test suite that hits this is not a trivial task. Thus I have deferred that part for now, and will open a new issue for further discussion.

**Comment by seanknox** [comment]
Thanks much @wbond!
